sale_delivery_date/hooks.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from openupgradelib import openupgrade
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+
+MERGED_MODULE_NAMES = (
+    "sale_warehouse_calendar",
+    "sale_cutoff_time_delivery",
+    "sale_partner_delivery_window",
+    "sale_partner_cutoff_delivery_window",
+)
+
+
+def pre_init_hook(cr):
+    """Since 4 modules (sale_warehouse_calendar, sale_cutoff_time_delivery,
+    sale_partner_delivery_window, sale_partner_cutoff_delivery_window)
+    are merged into this one, we have to set the previous modules as
+    uninstalled, as well as we need to update the module names in ir_model_data
+    """
+    modules_to_merge = [
+        (merged_module_name, "sale_delivery_date")
+        for merged_module_name in MERGED_MODULE_NAMES
+    ]
+    openupgrade.update_module_names(cr, modules_to_merge, merge_modules=True)
```

Comment:
You have to check first if the addons were installed in the database before calling `openupgrade`. This way Travis (which hasn't previous 13.0 installed) will installed your addon without issue.
For those having previous 13.0 addons installed in their DB, it will crash if `openupgradelib` is not installed, but that's fine in a migration process (we do not want to add `openupgradelib` in the external Python dependencies I think).